### PR TITLE
[Merged by Bors] - Use broker to query sync state

### LINKED ISSUE #1745: Handle not-synced in hare.go using broker.go
"no blocks for consensus" is a message we see in hare.go when we are not synced.
To handle it temporarily we suggested having hare.go hold the "isSynced" func and asking before printing it.
It is, however, not a good solution since the broker and the hare.go may now differ. We should provide a function in the broker that can be called from outside SAFELY and check for sync status.

### PR DESCRIPTION
**Pull request by gavraz**
## Motivation
We let the hare wrapper query for sync state directly.
This was done temporarily in order to have better logs.
The better way to do this is to ask the broker for sync state instead.

Closes #1745 

## Changes
Hare wrapper does not query sync state.
Broker exposes a query to answer sync state.

## Test Plan
Added UT for the broker.

### CODE REVIEW

**Comment by gavraz** [comment]
bors try

**Comment by gavraz** [comment]
bors try

File: tests/sync/test_sync.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def test_sync_gradually_add_nodes(init_session, setup_bootstrap, save_log_on_exi
 
     start = time.time()
     sleep = 30  # seconds
-    num_iter = 20  # total of 5 minutes
+    num_iter = 25  # total of 5 minutes
```
Comment (noamnelke) [inline tests/sync/test_sync.py:102]: intentional?

**Review by noamnelke — APPROVED** [review]

File: tests/sync/test_sync.py (hunk 1 above)
Comment (gavraz) [reply tests/sync/test_sync.py:102]: Yes, Almog said the sync test was a bit broken that we should have given it a bit more time

**Comment by gavraz** [comment]
bors merge

**Pull request closed without merging**